### PR TITLE
DRIVERS-291 Add collation option to Index management spec

### PR DESCRIPTION
**Pull request by estolfo**

### CODE REVIEW

File: source/index-management.rst [hunk 1]
```diff
@@ -669,6 +669,13 @@ Common API Components
      * filter expression are included in the index.
      */
     partialFilterExpression: Document;
+
+    /**
+     * Optionally specifies a collation to use for the index. If not specified, the default collation
```
Comment (behackett) [inline source/index-management.rst]: I don't think we want to use the collation specified on the collection, do we? It seems like the user should be explicit, as with all other options.

Comment (estolfo) [inline source/index-management.rst]: If a user creates a collection object with a default collation set, you don't think the user would expect an index created via that collection object would apply its default collation?

Comment (rozza) [inline source/index-management.rst]: I think I would expect the following from the Java API: 

* If a user doesn't provide any options when creating an index, I would create the options object with the collections collation settings.
* If a user provides options, then there wouldn't be any default for the collation settings, just what the user has set.

Comment (behackett) [inline source/index-management.rst]: If you create a collection (using the 'create' command) specifying a collation, will mongod use that collation when creating indexes for that collection?

Comment (estolfo) [inline source/index-management.rst]: I'm reading in the spec document that, upon index creation:

The absence of the collation parameter means that the index will use the collection’s collation. The system will automatically populate the index catalog’s persistent metadata infoObj with the inherited collation.

@dstorch can you confirm that an index created on a collection will use the collection's default collation in mongod, if it has one?

Comment (dstorch) [inline source/index-management.rst]: @estolfo, correct, that is the intended server behavior as currently specified.

Comment (behackett) [inline source/index-management.rst]: In that case, I retract my original comment. Drivers should behave the same as the server. Thanks!

Comment (estolfo) [inline source/index-management.rst]: ok, thanks for your review @behackett

File: source/index-management.rst [hunk 2]
```diff
@@ -669,6 +669,13 @@ Common API Components
...
+     * of the driver collection object is used, if one is defined. Otherwise, the default collation of
+     * the collection server-side is used.
```
Comment (behackett) [inline source/index-management.rst]: Also mention the server version this can be used with. See, for example, storageEngine.

Comment (estolfo) [inline source/index-management.rst]: ok, updated it.

**Comment by behackett** [comment]
LGTM

**Pull request merged**